### ISSUE TITLE
Minify CSS class names and ids

### ISSUE THREAD

**Issue by fstanis** [posthtml/htmlnano#69]
This can be done simply by calling [posthtml-uglify](https://github.com/Rebelmail/posthtml-uglify) inside `minifyCss`.

Bonus is that it can share the whitelist with uncss if we add uncss per #36.

**Comment by maltsev** [comment]
Good idea! I guess it might break some pages, so it's better to add it to [max preset](https://github.com/posthtml/htmlnano/blob/master/lib/presets/max.es6).

Does _posthtml-uglify_ take into account external styles? If not then we should do that on our side.

**Comment by fstanis** [comment]
> Does *posthtml-uglify* take into account external styles?

No, it only looks at `<style>` tags:

https://github.com/Rebelmail/posthtml-uglify/blob/fbf7a4ebf40481f092c060670ac8a5574421f05f/lib/plugin.js#L145-L154

> If not then we should do that on our side.

How would that work? Wouldn't that mean that `htmlnano` would have to modify files outside the HTML?

**Comment by maltsev** [comment]
> How would that work?

I see a few possible ways:

- Ignore classes and IDs found in external CSS. Not sure how easy it'd be to parse them, though.
- Rewrite names in external CSS. It seems a bit out of _htmlnano's_ scope for me, though.
- Disable minification at all if external CSS is found.

In an ideal case, we could implement all 3 options and let users configure that themselves how they want this minification work.

**Comment by fstanis** [comment]
Ouch, completely forgot that this will break external stylesheets. That's a tricky problem to solve.

> Disable minification at all if external CSS is found.

This seems like a good low-hanging fruit for a first version.

I'm also worried that [posthtml-uglify](https://github.com/Rebelmail/posthtml-uglify) seems unmaintained (at least based on PR / issue activity). Taking this into account, I'm now leaning more towards replicating similar functionality in htmlnano (or maybe forking the project?).

**Comment by maltsev** [comment]
> I'm also worried that _posthtml-uglify_ seems unmaintained (at least based on PR / issue activity).

I think it's totally fine to use it. We can always switch _htmlnano_ to a forked version if needed. We could also ask then if @Rebelmail are willing to transfer _posthtml-uglify_ to [posthtml](https://github.com/posthtml) namespace.

**Comment by polarathene** [comment]
You may also want to consider [`posthtml-minify-classnames`](https://github.com/posthtml/posthtml-minify-classnames/issues/1) which still receives maintenance and has a few more improvements over `posthtml-uglify`?
